### PR TITLE
Avoid double msgpack serialization of HLGs

### PR DESCRIPTION
**Pull request by madsbk**
`HighLevelGraph.__dask_distributed_pack__()` would explicit serialize using `dumps_msgpack()` instead of relying on implicit serialization. The result was double serialzaion of the high level graphs and large bytearrays in communication frames. 

Also, this PR makes `dumps_msgpack()` and `loads_msgpack()` obsolete and can be removed.

Fixes https://github.com/dask/distributed/issues/4652/

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

File: dask/highlevelgraph.py [hunk 1]
```diff
@@ -961,10 +961,10 @@ def __dask_distributed_pack__(self, client, client_keys: Iterable[Hashable]) ->
                     "annotations": layer.__dask_distributed_annotations_pack__(),
                 }
             )
-        return dumps_msgpack({"layers": layers})
+        return {"layers": layers}
```
Comment (gjoseph92) [inline dask/highlevelgraph.py:964]: This isn't important, but is there a msgpack/serialization reason for wrapping in a dict? Versus:
```suggestion
        return layers
```

**Review by gjoseph92 — COMMENTED** [review]
LGTM, also this seems like a key first step for https://github.com/dask/distributed/issues/4673

File: dask/highlevelgraph.py (hunk 1 above)
Comment (jrbourbeau) [reply dask/highlevelgraph.py:964]: This is a good point. I think we can probably just pass the list of layers directly. @madsbk mentioned that serializing twice is blocking some folks, so I'm going to merge this as is and we can address the dict vs. list point in a follow-up PR

**Review by jrbourbeau — APPROVED** [review]
Thanks @madsbk! I agree utilizing `distributed`s built-in serialization is a better way to go

File: dask/highlevelgraph.py (hunk 1 above)
Comment (madsbk) [reply dask/highlevelgraph.py:964]: The idea here was to make it easy to extend in the future but yes, changing to a list of layers is also fine.

**Pull request merged**